### ISSUE TITLE
Cannot pass back in point type as param

### ISSUE THREAD

**Issue by jexp** [neo4j-contrib/spatial#278]
If returning a new point from Cypher, the same object cannot be passed back in as a param to another Cypher call. We have fixed the mapping of internal types to external for returning objects, but not fixed the reverse for passing in params.

```
        ResourceIterator<Object> geometries = db.execute("RETURN point({latitude: 5.0, longitude: 4.0}) as geometry").columnAs("geometry");
        Geometry geom = (Geometry) geometries.next();
        geometries.close();
        System.out.println(geom);
        ResourceIterator<Object> results = db.execute("RETURN distance({geom}, point({latitude: 5.0, longitude: 4.0})) as distance", map("geom", geom)).columnAs("distance");
        Double distance = (Double) results.next();
        results.close();
        System.out.println(distance);
```

This code will fail with the error:

```
org.neo4j.cypher.internal.frontend.v3_0.CypherTypeException:
    Expected two Points, but got
        org.neo4j.cypher.internal.compatibility.helpersv3_0$$anon$2@750a04ec
    and
        GeographicPoint(4.0,5.0,CRS(WGS-84,4326,http://spatialreference.org/ref/epsg/4326/))

	at org.neo4j.cypher.internal.compiler.v3_0.commands.expressions.DistanceFunction.apply(DistanceFunction.scala:48)
...
```

**Comment by jexp** [comment]
Need to make the public-private mapper symmetrical for both Point and Geometry types. This is so that neo4j-spatial does not need to deal with all combinations of public-private types, hopefully only public types.

**Comment by jexp** [comment]
Need to make the public-private mapper symmetrical for both Point and Geometry types. This is so that neo4j-spatial does not need to deal with all combinations of public-private types, hopefully only public types.

**Comment by craigtaverner** [comment]
I believe this is fixed in all versions since Neo4j 3.4
